Documentation/BasicUsage.md

```diff
@@ -0,0 +1,238 @@
+window.MediaStreamTrack
+```
+
+## Defining Media Constraints
+
+```javascript
+let mediaConstraints = {
+	audio: true,
+	video: {
+		frameRate: 30,
+		facingMode: 'user'
+	}
+};
+```
+
+## Getting a Media Stream
+
+```javascript
+let localMediaStream;
+let isVoiceOnly = false;
+
+try {
+	const mediaStream = await mediaDevices.getUserMedia( mediaConstraints );
+
+	if ( isVoiceOnly ) {
+		let videoTrack = await mediaStream.getVideoTracks()[ 0 ];
+		videoTrack.enabled = false;
+	};
+
+	localMediaStream = mediaStream;
+} catch( err ) {
+	// Handle Error
+};
+```
+
+## Destroying a Media Stream
+
+```javascript
+localMediaStream.getTracks().map(
+	track => track.stop()
+);
+
+localMediaStream = null;
+```
+
+## Defining Peer Constraints
+
+```javascript
+let peerConstraints = {
+	iceServers: [
+		{
+			url: 'stun:stun.l.google.com:19302'
+		}
+	]
+};
+```
+
+## Creating a Peer Connection
+
+```javascript
+let peerConnection = new RTCPeerConnection( peerConstraints );
+
+peerConnection.onicecandidate = handleLocalCandidate;
+peerConnection.onicecandidateerror = handleCandidateError;
+peerConnection.oniceconnectionstatechange = handleICEConnectionStateChange;
+peerConnection.onconnectionstatechange = handleConnectionStateChange;
+peerConnection.onsignalingstatechange = handleSignalingStateChange;
+peerConnection.onnegotiationneeded = handleNegotiation;
+peerConnection.onaddstream = handleStreamAdded;
+peerConnection.onremovestream = handleStreamRemoved;
+```
+
+## Destroying a Peer Connection
+
+```javascript
+peerConnection.onicecandidate = null;
+peerConnection.onicecandidateerror = null;
+peerConnection.oniceconnectionstatechange = null;
+peerConnection.onconnectionstatechange = null;
+peerConnection.onsignalingstatechange = null;
+peerConnection.onnegotiationneeded = null;
+peerConnection.onaddstream = null;
+peerConnection.onremovestream = null;
+
+peerConnection._unregisterEvents();
+peerConnection.close();
+peerConnection = null;
+```
+
+## Adding a Data Channel
+
+```javascript
+
+```
+
+## Adding a Media Stream
+
+```javascript
+peerConnection.addStream( localMediaStream );
+```
+
+## Dealing with ICE Candidates and Peer Events
+
+```javascript
+function handleLocalCandidate( event ) {
+	// If we've reached the end, don't send anything.
+	if ( !event.candidate ) { return; };
+
+	// Send the ICE Candidate to the call recipient.
+};
+
+function handleCandidateError( err ) {
+	// Handle Error
+};
+
+function handleICEConnectionStateChange( event ) {
+
+};
+
+function handleConnectionStateChange( event ) {
+
+};
+
+function handleSignalingStateChange( event ) {
+
+};
+
+function handleNegotiation() {
+
+};
+
+function handleStreamAdded( event ) {
+
+};
+
+function handleStreamRemoved( event ) {
+
+};
+```
+
+## Defining Session Constraints
+
+```javascript
+let sessionConstraints = {
+	mandatory: {
+		OfferToReceiveAudio: true,
+		OfferToReceiveVideo: true,
+		VoiceActivityDetection: true
+	},
+	optional: [
+		{
+			DtlsSrtpKeyAgreement: true
```

Comment:
Drop this, it's no longer optional since SDES support has been removed.

Reply:
Removed in the next.